src/main/webapp/app/shared/competency-selection/competency-selection.component.ts

```diff
@@ -21,14 +22,16 @@ import { finalize } from 'rxjs';
 export class CompetencySelectionComponent implements OnInit, ControlValueAccessor {
     @Input() labelName: string;
     @Input() labelTooltip: string;
-    @Input() value: any;
+    // selected competencies
+    @Input() value?: Competency[];
```

Comment:
could rename to selectedCompetencies, also could avoid the undefined checks by setting its default value to `[ ]`